src/providers/irc/IrcConnection2.cpp

```diff
@@ -64,7 +68,25 @@ IrcConnection::IrcConnection(QObject *parent)
                 // If we're still receiving messages, all is well
                 this->recentlyReceivedMessage_ = false;
                 this->waitingForPong_ = false;
-                this->heartbeat.invoke();
+
+                // Check if we got invoked too late (e.g. due to a sleep)
+                auto now = std::chrono::system_clock::now();
+                auto elapsed = now - this->lastPing_;
```

Comment:
Since `lastPing_` is default-initialized to 0, the first time we get here, we will skip one heartbeat skipped

Reply:
Shouldn't be a major issue.